protocol/options.go

```diff
@@ -27,12 +28,30 @@ import (
 
 type Options struct {
 	Protocol *global.ProtocolConfig
+
+	ID string
 }
 
-func DefaultOptions() *Options {
+func defaultOptions() *Options {
 	return &Options{Protocol: global.DefaultProtocolConfig()}
 }
 
+func NewOptions(opts ...Option) *Options {
+	defOpts := defaultOptions()
+	for _, opt := range opts {
+		opt(defOpts)
+	}
+
+	if defOpts.Protocol.Name == "" {
+		panic(fmt.Sprintf("Please specify registry, eg. WithTriple()"))
```

Comment:
Should we default to 'triple' protocol if the protocol name is not present?

So, this raises another question, what's the current default behavior if users do not specify any information about the protocol at all? 

``server.NewServer()`

Reply:
For unifying design principle of API, setting default value to **triple** protocol and deleting ```WithTriple()``` make sense. But the consideration is the same as registry configuration.
For instance, setting **zookeeper** as default registry is not intuitive to users because **zookeeper** is not a de facto standard. Users often need to be aware that the concrete registry they are using.
Likewise, **triple** is not a well-known protocol so that users need to configure it clearly.

But now I think setting **triple** as default protocol could help us promote it. **triple** could replace **http** and **grpc** in dubbo ecosphere. These is only **dubbo** protocol that we need to take into separate consideration.

Reply:
If users do not specify any information about the protocol at all, server would not serve any protocol. Maybe we should raise an error when users do not provide any protocol configurations when initializing **ServerOptions**?